docs/guides/latest/workloads/daemonset.md

```diff
@@ -0,0 +1,759 @@
+NAME                READY   STATUS    RESTARTS   AGE
+stash-demo-c4nqw    1/1     Running   0          39s
+```
+
+Verify that the sample data has been created in `/source/data` directory using the following command,
+
+```console
+$ kubectl exec -n demo stash-demo-c4nqw -- ls -R /source/data
+/source/data:
+sample-file.txt
+```
+
+### Prepare Backend
+
+We are going to store our backed up data into a GCS bucket. At first, we need to create a secret with GCS credentials then we need to create a Repository crd. If you want to use a different backend, please read the respective backend configuration doc from [here](/docs/guides/latest/backends/overview.md).
+
+**Create Secret:**
+
+Let's create a secret called `gcs-secret` with access credentials to our desired GCS bucket,
+
+```console
+$ echo -n 'changeit' > RESTIC_PASSWORD
+$ echo -n '<your-project-id>' > GOOGLE_PROJECT_ID
+$ cat downloaded-sa-json.key > GOOGLE_SERVICE_ACCOUNT_JSON_KEY
+$ kubectl create secret generic -n demo gcs-secret \
+    --from-file=./RESTIC_PASSWORD \
+    --from-file=./GOOGLE_PROJECT_ID \
+    --from-file=./GOOGLE_SERVICE_ACCOUNT_JSON_KEY
+secret/gcs-secret created
+```
+
+**Create Repository:**
+
+Now, create a `Respository` using this secret.
+
+Below is the YAML of `Repository` crd we are going to create,
+
+```console
+apiVersion: stash.appscode.com/v1alpha1
+kind: Repository
+metadata:
+  name: gcs-repo
+  namespace: demo
+spec:
+  backend:
+    gcs:
+      bucket: appscode-qa
+      prefix: /source/data/sample-daemonset
+    storageSecretName: gcs-secret
+```
+
+Let's create the Repository we have shown above,
+
+```console
+$ kubectl apply -f ./docs/examples/workloads/daemonset/repository.yaml
+repository.stash.appscode.com/gcs-repo created
+```
+
+Now, we are ready to backup our volumes to our desired backend.
+
+### Backup
+
+We have to create a `BackupConfiguration` crd targeting the `stash-demo` Daemonset that we have deployed earlier. Then Stash will inject a sidecar container to the target. It will also create a `CronJob` to take periodic backup of `/source/data` directory of the target.
+
+**Create BackupConfiguration:**
+
+Below is the YAML of the `BackupConfiguration` crd that we are going to create,
+
+```console
+apiVersion: stash.appscode.com/v1beta1
+kind: BackupConfiguration
+metadata:
+  name: dmn-backup
+  namespace: demo
+spec:
+  repository:
+    name: gcs-repo
+  schedule: "*/1 * * * *"
+  target:
+    ref:
+      apiVersion: apps/v1
+      kind: DaemonSet
+      name: stash-demo
+    volumeMounts:
+    - name: source-data
+      mountPath: /source/data
+    directories:
+    - /source/data
+  retentionPolicy:
+    name: 'keep-last-5'
+    keepLast: 5
+    prune: true
+```
+
+Here,
+
+- `spec.repository` refers to the `gcs-repo` GCP Backend.
+- `spec.schedule` is a cron expression indicates that `BackupSession` will be created at 1 minute interval.
+- `spec.target.ref` refers to the target workload that was created for `stash-demo` Daemonset.
+
```

Comment:
- `spec.repository` refers to the `Repository` object `gcs-repo` that holds backend information.
- `spec.schedule` is a cron expression that indicates `BackupSession` will be created at 1 minute interval.
- `spec.target.ref` refers to the `stash-demo` DaemonSet.